tools/trap/trap.c

```diff
@@ -288,6 +289,67 @@ static int extfsdump_recv(int dev_fd)
         return 0;
 }
 
+int dump_file_from_fs(int dev_fd, char *filename)
+{
+	int ret;
+	int fd;
+	int i, last_subdir;
+	char buf;
+	FILE *fp;
+	u_int64_t file_size;
+	char char_filesize[8];
+	char local_filename[CONFIG_SMARTFS_MAXNAMLEN + 9];
+
+	printf("Attempting to Dump File %s from Target Device\n", filename);
+
+	ret = write(dev_fd, filename, CONFIG_SMARTFS_MAXNAMLEN);
+	if (ret != CONFIG_SMARTFS_MAXNAMLEN) {
+		printf("Unable to send Dump File name to Target Device\n");
+		return -1;
+	}
+
+	file_size = 0;
+	for (int i = 0; i < 8; i++) {
+		ret = read(dev_fd, &char_filesize[i], sizeof(char_filesize[i]));
+		if (ret != sizeof(char_filesize[i])) {
+			printf("Unable to get file size from Target Device\n");
+			return -1;
```

Comment:
please check for consistency of return statements in all places. Some places there is flag based return, and in some there is numeric returns.

Reply:
This is already under discussion with regards to line 254 in one of the above conversations.
Please check that